lib/thrift/protocol/binary.ex

```diff
@@ -10,35 +10,24 @@ defmodule Thrift.Protocol.Binary do
 
   alias Thrift.TApplicationException
 
-  @typedoc "Binary protocol field type identifier"
-  @type type_id :: 2 | 3 | 4 | 6 | 8 | 10 | 11 | 12 | 13 | 14 | 15
+  require Thrift.Protocol.Binary.Type, as: Type
+
   @type deserializable :: :message_begin | :application_exception
 
-  @stop   0
-  @bool   2
-  @byte   3   # also: i8
-  @double 4
-  @i16    6
-  @i32    8
-  @i64    10
-  @string 11
-  @struct 12
-  @map    13
-  @set    14
-  @list   15
-
-  @spec type_id(Thrift.data_type) :: type_id
-  defp type_id(:bool),      do: @bool
-  defp type_id(:byte),      do: @byte
-  defp type_id(:i16),       do: @i16
-  defp type_id(:i32),       do: @i32
-  defp type_id(:i64),       do: @i64
-  defp type_id(:double),    do: @double
-  defp type_id(:string),    do: @string
-  defp type_id(:struct),    do: @struct
-  defp type_id({:map, _}),  do: @map
-  defp type_id({:set, _}),  do: @set
-  defp type_id({:list, _}), do: @list
+  @stop 0
+
+  @spec type_id(Thrift.data_type) :: Type.t
+  defp type_id(:bool),      do: Type.bool
+  defp type_id(:byte),      do: Type.byte
+  defp type_id(:i16),       do: Type.i16
+  defp type_id(:i32),       do: Type.i32
+  defp type_id(:i64),       do: Type.i64
+  defp type_id(:double),    do: Type.double
+  defp type_id(:string),    do: Type.string
+  defp type_id(:struct),    do: Type.struct
+  defp type_id({:map, _}),  do: Type.map
+  defp type_id({:set, _}),  do: Type.set
+  defp type_id({:list, _}), do: Type.list
```

Comment:
I thought of doing that and wanted to get feedback on the initial approach first.

One "awkward" aspect of function in `struct_binary_protocol.ex` is that they use generator-specific types (e.g. `%Union{}`), and referencing those types from `Thrift.Protocol.` seems a little yucky.